velox/connectors/hive/storage_adapters/gcs/tests/GCSFileSystemTest.cpp

```diff
@@ -0,0 +1,353 @@
+      error += " " + interpreter;
+      if (exe_path.empty()) {
+        error += " (exe not found)";
+        continue;
+      }
+
+      server_process_ = bp::child(
+          boost::this_process::environment(),
+          exe_path,
+          "-m",
+          "testbench",
+          "--port",
+          port_,
+          group_);
+      if (server_process_.valid() && server_process_.running())
+        break;
+      error += " (failed to start)";
+      server_process_.terminate();
+      server_process_.wait();
+    }
+    if (server_process_.valid() && server_process_.valid())
+      return;
+    error_ = std::move(error);
+  }
+
+  ~GcsTestbench() override {
+    // Brutal shutdown, kill the full process group because the GCS testbench
+    // may launch additional children.
+    group_.terminate();
+    if (server_process_.valid()) {
+      server_process_.wait();
+    }
+  }
+
+  const std::string& port() const {
+    return port_;
+  }
+  const std::string& error() const {
+    return error_;
+  }
+
+ private:
+  std::string port_;
+  bp::child server_process_;
+  bp::group group_;
+  std::string error_;
+};
+
+using namespace facebook::velox;
+
+class GCSFileSystemTest : public testing::Test {
+ protected:
+  static void SetUpTestSuite() {
+    if (testbench_ == nullptr) {
+      testbench_ = std::make_shared<GcsTestbench>();
+    }
+    filesystems::registerGCSFileSystem();
+
+    ASSERT_THAT(testbench_, ::testing::NotNull());
+    ASSERT_THAT(testbench_->error(), ::testing::IsEmpty());
+
+    // Create a bucket and a small file in the testbench. This makes it easier
+    // to bootstrap GcsFileSystem and its tests.
+    auto client = gcs::Client(
+        google::cloud::Options{}
+            .set<gcs::RestEndpointOption>(
+                "http://localhost:" + testbench_->port())
+            .set<gc::UnifiedCredentialsOption>(gc::MakeInsecureCredentials()));
+
+    bucket_name_ = "test1-gcs";
+    google::cloud::StatusOr<gcs::BucketMetadata> bucket =
+        client.CreateBucketForProject(
+            bucket_name_, "ignored-by-testbench", gcs::BucketMetadata{});
+    ASSERT_TRUE(bucket.ok()) << "Failed to create bucket <" << bucket_name_
+                             << ">, status=" << bucket.status();
+
+    object_name_ = "test-object-name";
+    google::cloud::StatusOr<gcs::ObjectMetadata> object =
+        client.InsertObject(bucket_name_, object_name_, kLoremIpsum);
+    ASSERT_TRUE(object.ok()) << "Failed to create object <" << object_name_
+                             << ">, status=" << object.status();
+  }
+
+  std::shared_ptr<const Config> TestGcsOptions() const {
+    std::unordered_map<std::string, std::string> configOverride = {};
+
+    configOverride["hive.gcs.scheme"] = "http";
+    configOverride["hive.gcs.endpoint"] = "localhost:" + testbench_->port();
+    return std::make_shared<const core::MemConfig>(std::move(configOverride));
+  }
+
+  std::string PreexistingBucketName() {
```

Comment:
camelCase for function names https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md#naming-conventions

Reply:
Agree, I will fix it.